### PR TITLE
[AMORO-2198][Dashboard] Fixing the parameter misuse when calling the branches/tags HTTP API

### LINKED ISSUE #2198: [Subtask]: [Dashboard] Support filters of `Snapshots` page for table
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

Support filters for `Snapshots` page. As desceibed in #2034 , we are going to support two filters in the `Snapshots` page.

image

**Branches/Tags Filter:**
- the tags/branches in the list are from the HTTP API introduced from https://github.com/NetEase/amoro/pull/2172
- the HTTP API will return at least 1 branch
- the first branch of the list should be the default value of the filter
- only support select 1 branch or tag at a time

**Operations Filter:**
- the list contains 3 fixed items, `all`,`optimizing`,`non-optimizing`
- `all` is the default value
- after swith a tag/branch, the operations filter should change to select the default value `all`
- only support select 1 operation at a time

Both filters will task effects on the HTTP API `.../tables/{table}/snapshots` in the format of: 
`.../tables/{table}/snapshots?ref={tag/branch}&operation={all}`

Besides, we also have to support displying how many branches and tags there are, which also comes from this HTTP API
- https://github.com/NetEase/amoro/pull/2172.

### Parent issue

- #1354

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

Close #2198 .

## Brief change log

- fix: the catalog name should not be a fixed value

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / **no**)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Comment by wangtaohz** [comment]
Previously, testing with different catalog names was missed. This is the test result after the fix.

image

**Review by minteliuwm — APPROVED** [review]
LGTM

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java [hunk 1]
```diff
@@ -607,6 +607,9 @@ public void getTableBranches(Context ctx) {
   }
 
   private void putMainBranchFirst(List<TagOrBranchInfo> branchInfos) {
+    if (branchInfos.size() <= 1) {
+      return;
```
[the remark this answers is not in the document]
Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java:611]: First, when the list has only one element, there is no need to reorder it. 

Second, for formats that do not support branches (such as Paimon, Mixed Format), here a immutable list with only a `main` branch is returned, disallowing reordering.

**Review by HuangFru — APPROVED** [review]
I also encountered this problem, thanks for the fix, LGTM.

**Review by baiyangtx — APPROVED** [review]
LGTM

**Comment by tcodehuber** [comment]
LGTM

**Pull request merged**